include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -135,29 +149,40 @@ struct __parallel_transform_reduce_small_submitter
     }
 }; // struct __parallel_transform_reduce_small_submitter
 
-// General parallel_transform_reduce - uses a tree-based reduction
-template <typename _Tp, typename _IsGPU>
-struct __parallel_transform_reduce_submitter
+template <::std::uint16_t __work_group_size, ::std::size_t __iters_per_work_item, typename _Tp, typename _ReduceOp,
+          typename _TransformOp, typename _ExecutionPolicy, typename _Size, typename _InitType,
+          oneapi::dpl::__internal::__enable_if_device_execution_policy<_ExecutionPolicy, int> = 0, typename... _Ranges>
+auto
+__parallel_transform_reduce_small_impl(_ExecutionPolicy&& __exec, _Size __n, _ReduceOp __reduce_op,
+                                       _TransformOp __transform_op, _InitType __init, _Ranges&&... __rngs)
 {
-    template <typename... _Name>
-    using _KernelName = __reduce_kernel<_IsGPU{}, _Name...>;
+    using _Policy = typename ::std::decay<_ExecutionPolicy>::type;
+    using _CustomName = typename _Policy::kernel_name;
+    using _ReduceKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __reduce_small_kernel<__work_group_size, __iters_per_work_item, _CustomName>>;
 
-    template <typename _ExecutionPolicy, typename _Size, typename _ReduceOp, typename _TransformOp1,
-              typename _TransformOp2, typename _InitType,
+    return __parallel_transform_reduce_small_submitter<__work_group_size, __iters_per_work_item, _Tp, _ReduceKernel>()(
+        ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
+        ::std::forward<_Ranges>(__rngs)...);
+}
+
+// General parallel_transform_reduce - uses a tree-based reduction
+template <typename _Tp, typename _IsGPU, typename _KernelName>
+struct __parallel_transform_reduce_submitter;
+
+template <typename _Tp, typename _IsGPU, typename... _Name>
+struct __parallel_transform_reduce_submitter<_Tp, _IsGPU, __internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _ExecutionPolicy, typename _ReduceOp, typename _TransformOp1, typename _TransformOp2,
+              typename _Size, typename _InitType,
               oneapi::dpl::__internal::__enable_if_device_execution_policy<_ExecutionPolicy, int> = 0,
               typename... _Ranges>
-    static auto
-    submit(_ExecutionPolicy&& __exec, _Size __n, ::std::uint16_t __work_group_size, _ReduceOp __reduce_op,
-           _TransformOp1 __transform_op1, _TransformOp2 __transform_op2, _InitType __init, _Ranges&&... __rngs)
+    auto
+    operator()(_ExecutionPolicy&& __exec, _Size __n, ::std::uint16_t __work_group_size, _ReduceOp __reduce_op,
```

Comment:
The problem is we cannot use submitter in places where we also use kernel_bundle and kernel compiler

Reply:
Thanks, I have reverted the changes for the compiled kernel using the kernel bundle.